fix: CacheChecker filters syntax

vedantnaik19 opened the pull request:
### Related Issues

- fixes #7892

### Proposed Changes:

-  Use new filter syntax in the `CacheChecker` component instead of legacy ones.

### How did you test it?

- Existing unit tests

### Notes for the reviewer

N/A

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack/blob/main/code_of_conduct.txt)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.
- I documented my code
- I ran [pre-commit hooks](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md#installation) and fixed any issue

shadeMe reviewed (dismissed):
Thanks for the PR! Could you add the release note and a unit test?

vedantnaik19 commented:
> Thanks for the PR! Could you add the release note and a unit test?

Hello 👋 
I have added a release note now. 
However, I am not sure if I should add another unit test, as there is an existing test which checks the functionality. 🤔

silvanocerza commented:
> Hello 👋 I have added a release note now. However, I am not sure if I should add another unit test, as there is an existing test which checks the functionality. 🤔

@vedantnaik19 the current test is not that great. Best thing would be to use a mocked Document Store and verify that `filter_documents()` is called with the correct filters syntax. 

As of now it's indirectly testing `InMemoryDocumentStore.filter_documents()` and that's why we didn't notice it was using the old filters syntax.

vedantnaik19 commented:
> > Hello 👋 I have added a release note now. However, I am not sure if I should add another unit test, as there is an existing test which checks the functionality. 🤔
> 
> @vedantnaik19 the current test is not that great. Best thing would be to use a mocked Document Store and verify that `filter_documents()` is called with the correct filters syntax.
> 
> As of now it's indirectly testing `InMemoryDocumentStore.filter_documents()` and that's why we didn't notice it was using the old filters syntax.

Hello 👋 
I have added a test now. Please let me know if this is OK.
Also, thanks for your previous feedback @silvanocerza @shadeMe 🙌

silvanocerza reviewed (approved):
Looks good, thanks!
